### PR TITLE
Updated test_factorial in Python. Issue '#1621'

### LINKED ISSUE #1621: `test_factorial_invalid` should use `invalid_permutations`
`test_factorial_invalid` in test_factorial currently uses `valid_permutations`. It should be using `invalid_permutations`.

It needs to be changed to use the correct permutations.
Then, any tests that break from this change need to be fixed.

### PR DESCRIPTION
**Pull request by 13point5**
## Complete the Applicable Sections Below

Find which section best describes your pull request and make sure you fill it out. To start, let us know which issue you've fixed.

- [x] I fixed #1621 

### Code Snippets

- [x] I named the pull request using `Added/Updated <Sample Program> in <Language>` format

## Notes
Replaced the `valid_permutations` parameter with `invalid_permutations` in `test/projects/test_factorial.py`

### CODE REVIEW

**Comment by auroq** [comment]
Hey @13point5. Thank you for taking this on! The change to the test looks good, but several tests are failing now because the code snippets don't give expected output on invalid input.

See build here:
https://travis-ci.com/TheRenegadeCoder/sample-programs/jobs/245787435

Are you able to fix these?

Looks like there are issues in
- factorial.py
- factorial.c
- factorial.cs
- factorial.java
- factorial.js

**Comment by auroq** [comment]
Aslo, if there's just a few of them that you don't feel comfortable with, let me know and I can handle those while you do the rest.

**Comment by 13point5** [comment]
Yes, I will try to fix them. I'm new to this so I will gladly accept your help @auroq , Thank You!

**Comment by 13point5** [comment]
I would like to know how I can run these tests on my machine. Can you help me with this?

**Comment by auroq** [comment]
First you will need both docker and python 3 installed.
Then run the test by using
`python -um samplerunner test -p factorial`

That will run all the factorial tests.

Use 
`python -um samplerunner test -p factorial.java`
to run the tests for java or replace `factorial.java` with any snippet to run those.

**Comment by auroq** [comment]
Note that docker hub is having downtime at the moment. Unless you have the docker images locally already, you won't be able to run tests until this is resolved:

https://status.docker.com/pages/incident/533c6539221ae15e3f000031/5da5fc78c7b1c11cd8603721

**Comment by 13point5** [comment]
Okay, I haven't used docker previously but if I run `python -um
samplerunner download` then will I get them?

**Comment by auroq** [comment]
Doing `samplerunner test` will automatically download, so you don't need to explicitly download.

If you want to download everything all at once you could do `samplerunner download -p factorial`, but that will also download images you won't need for fixing this subset of tests.

Regardless, you will need to have docker installed and running on your machine.

**Comment by auroq** [comment]
@13point5, we just merged some big changes to our test runner. It doesn't change  much from the functionality, but how you call it changed slightly. I updated my comments above to show how it is called now. Let me know if you have any more questions.

**Comment by 13point5** [comment]
I'm sorry but I really need help with docker. I know I need to use the .travis.yml file but I don't know what to do with it. I have installed docker and I'm using Ubuntu 18.04

**Comment by auroq** [comment]
Awesome. A fellow linux user. :penguin:
I'm in the process of updating our documentation to make all of this easier, but the basic steps are:

1. Make sure both docker and python 3+ are installed.
2. Ensure your user is able to run docker commands (use `docker info` or `docker ps` to test) Follow instructions here if it doesn't work: https://docs.docker.com/install/linux/linux-postinstall/
3. `pip install -r requirements.txt` (preferably in a virtual environment)
4. `./samplerunner.sh test -p factorial` or `python runner.py test -p factorial`

File: archive/c/c/factorial.c [hunk 1]
```diff
@@ -15,7 +15,8 @@ int main (int argc, char *argv[])
 
 	/* Correct nmber of args? */
 	if (argc != 2) {
-		printf("Usage: %s [positive integer]\n", argv[0]);
+		// printf("Usage: %s [positive integer]\n", argv[0]);
```
Comment (auroq) [inline archive/c/c/factorial.c]:
```suggestion

```
I don't see a need to leave this commented. Let's just remove the line.

File: archive/c/c/factorial.c [hunk 2]
```diff
@@ -31,7 +32,8 @@ int main (int argc, char *argv[])
 	if (endptr == argv[1] ||
 			!(*endptr == '\0' || *endptr == '\n') ||
 			n < 0) {
-		fprintf(stderr, "Not a positive integer\n");
+		// fprintf(stderr, "Not a positive integer\n");
```
Comment (auroq) [inline archive/c/c/factorial.c]:
```suggestion

```
Same thing here. Let's just remove the line.

**Comment by jrg94** [comment]
I recently updated some the Elixir solutions. Looks like they made it into this branch, and now they're failing for a couple tests.

**Comment by auroq** [comment]
I'm going to merge this to get the Elixir changes from master. If there's issues once it's merged, we can fix it or revert it.

**Comment by 13point5** [comment]
Cool, I'll pick another issue tomorrow and start working on it.

**Pull request merged**